applications/rag/frontend/container/cloudbuild.yaml

```diff
@@ -0,0 +1,29 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# To build the base image:
+# 1) Run `gcloud config set project ai-on-gke`
+# 2) Change the Dockerfile in the args to 'Dockerfile.base'
+# 3) Run `gcloud builds submit --config cloudbuild.yaml --substitutions=_PROJECT="ai-on-gke",_LOCATION="us-central1",_REPOSITORY="rag-on-gke",_IMAGE="frontend" .` in the same directory.
+
+# To build the langchain image (note the difference in the image name):
+# 1) Run `gcloud config set project ai-on-gke`
+# 2) Change the Dockerfile in the args to 'Dockerfile.langchain'
+# 3) Run: `gcloud builds submit --config cloudbuild.yaml --substitutions=_PROJECT="ai-on-gke",_LOCATION="us-central1",_REPOSITORY="rag-on-gke",_IMAGE="frontend-langchain" .` in the same directory.
+
+steps:
+- name: 'gcr.io/cloud-builders/docker'
+  args: [ 'build', '-t', '${_LOCATION}-docker.pkg.dev/${_PROJECT}/${_REPOSITORY}/${_IMAGE}', '-f', 'Dockerfile.langchain', '.' ]
```

Comment:
same comment, maybe just 1 image?

Reply:
The value in 2 images is that the build time is much shorter for the second one.